File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    let vowels = ["a", "e", "i", "o", "u"]
    s = s.toLowerCase()
    let mid = Math.floor(s.length / 2)
    let count_1 = 0
    let count_2 = 0
    for(let i = 0; i < mid; i++){
        if(vowels.includes(s[i])) count_1++;
    }
    for(let i = mid; i < s.length; i++){
        if(vowels.includes(s[i])) count_2++;
    }
    return count_1 == count_2;
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        char[] a = s.toCharArray();
        int count1 = 0;
        int count2 = 0;
        int b = a.length/2;
        for(int i=0 ; i<b ; i++ ){
            if(a[i]=='a'||a[i]=='e'||a[i]=='i'||a[i]=='o'||a[i]=='u'||a[i]=='A'||a[i]=='E'||a[i]=='I'||a[i]=='O'||a[i]=='U'){
                count1++;
            }
        }
        for(int i=b ; i<a.length ; i++ ){
            if(a[i]=='a'||a[i]=='e'||a[i]=='i'||a[i]=='o'||a[i]=='u'||a[i]=='A'||a[i]=='E'||a[i]=='I'||a[i]=='O'||a[i]=='U'){
                count2++;
            }
        }
        if(count1==count2){
            return true;
        }
        return false;

    }
}